Structure_Engine/Compute/PanelToFEMesh.cs

```diff
@@ -0,0 +1,106 @@
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+
+using BH.oM.Structure.Elements;
+using BH.oM.Geometry;
+using BH.Engine.Geometry;
+using BH.oM.Reflection.Attributes;
+using System.Collections.Generic;
+using System.Linq;
+using System.ComponentModel;
+using BH.Engine.Analytical;
+
+namespace BH.Engine.Structure
+{
+    public static partial class Compute
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+        [Description("Converts a Panel with three or four control points to a FEmesh with a single Face. This is not a method to discretise a Panel, it simply converts a simple Panel to an identical FEMesh.")]
+        [Input("panel", "Panel to be converted to a FEMesh.")]
+        [Output("FEMesh", "FEMesh converted from a Panel.")]
+
+        public static FEMesh PanelToFEMesh(Panel panel)
+        {
+            List<Point> points = new List<Point>();
+            List<Edge> edges = panel.ExternalEdges;
+            List<Face> faces = new List<Face>();
+            if (NullCheck(panel) == false)
+            {
+                return null;
+            }
+            if(!panel.Geometry().IsPlanar(Tolerance.MacroDistance))
+            {
+                Reflection.Compute.RecordError("Panel is not planar and therefore cannot be converted to an FEMesh.");
+            }
+            if (panel.Openings.Count > 0)
+            {
+                Reflection.Compute.RecordError("This method does not support Panels with Openings");
+                return null;
+            }
+            if (edges.Count > 4)
+            {
+                Reflection.Compute.RecordError("Panel contains more than 4 Edges");
+                return null;
+            }
+            foreach (Edge edge in edges)
+            {
+                ICurve curve = Analytical.Query.Geometry(edge);
+                points.AddRange(Geometry.Convert.IToPolyline(curve).ControlPoints);
+            }
+            int Count = points.Distinct().Count();
+            Face face = new Face();
+            if (Count>4)
+            {
+                Reflection.Compute.RecordError("Panel contains more than 4 Control Points");
+                return null;
+            }
+            if (Count == 4)
+            {
+                face = Geometry.Create.Face(0, 1, 2, 3);
+            }
+            else if (Count == 3)
+            {
+                face = Geometry.Create.Face(0, 1, 2);
+            }
+            faces.Add(face);
+            Mesh mesh = Geometry.Create.Mesh(points.Distinct(), faces);
+            FEMesh fEMesh = new FEMesh();
+            if (panel.Property != null)
+            {
+                fEMesh = Create.FEMesh(mesh, panel.Property, null, panel.Name);
+                return fEMesh;
```

Comment:
As above on the `FEMeshToPanel` - create the `FEMesh` first, then check if the `SurfaceProperty` is `null`, then you can assign it or skip.